src/Core/Http/SecureHttpClient.php

```diff
@@ -392,9 +394,10 @@ private function createRequest($method, $endpoint, $data)
         $request = new Request($method, $endpoint);
         $request = $request->withHeader('Accept', 'application/json,application/jose+json,');
 
-        if ('POST' === $method && \is_array($data)) {
+        if ('POST' === $method) {
             $request = $request->withHeader('Content-Type', 'application/jose+json');
-            $request = $request->withBody(\GuzzleHttp\Psr7\stream_for(json_encode($data)));
+            $body = \GuzzleHttp\Psr7\stream_for(null === $data ? '' : json_encode($data));
+            $request = $request->withBody($body);
```

Comment:
variable `$body` is useless. `$request = $request->withBody(\GuzzleHttp\Psr7\stream_for(null === $data ? '' : json_encode($data)));`

moreover, I thing that you don't need to call withBody here
```php
if (null !== $data) {
    $request = $request->withBody(\GuzzleHttp\Psr7\stream_for(json_encode($data)));
}
```